### PR TITLE
test: update Jest to latest version, 29.7.0

### PR DESCRIPTION
**Pull request by EvanHahn**
*This change is not urgent.*

This updates `jest` and `@types/jest` to their latest versions.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -116,9 +116,9 @@
     "@babel/core": "^7.20.0",
     "@babel/preset-env": "^7.20.0",
     "@babel/runtime": "^7.20.0",
-    "@formatjs/cli": "^6.2.0",
     "@comapeo/core": "1.0.1",
     "@comapeo/schema": "1.0.0",
+    "@formatjs/cli": "^6.2.0",
```
Comment (EvanHahn) [inline package.json:121]: This line moved (`npm install` alphabetizes the list) but the dependency didn't actually change.

**Review by cimigree — COMMENTED** [review]
Should the package-lock.json be part of this PR?

**Comment by EvanHahn** [comment]
I think so, because the installed dependencies changed.

**Comment by cimigree** [comment]
> I think so, because the installed dependencies changed.

Oh ha ha it is! Don't know how I missed that! Sorry!

**Review by cimigree — APPROVED** [review]

**Pull request merged**